feat(ant-node): add relay_client to NodeSpawner and improve examples

vphongph opened the pull request:
- Add relay_client option to NodeSpawner for NAT traversal
- Add warning for zero rewards address in NodeSpawner/NetworkSpawner
- Add spawn_nodes_on_mainnet.rs example with relay and env var support
- Update spawn_local_network.rs with client connection and cost quote
- Add README.md documenting spawner APIs and options
- Remove obsolete spawn_node.rs example
- Other files: clippy formatting

vphongph committed:
feat(ant-node): add relay_client to NodeSpawner and improve examples

- Add relay_client option to NodeSpawner for NAT traversal
- Add warning for zero rewards address in NodeSpawner/NetworkSpawner
- Add spawn_nodes_on_mainnet.rs example with relay and env var support
- Update spawn_local_network.rs with client connection and cost quote
- Add README.md documenting spawner APIs and options
- Remove obsolete spawn_node.rs example
- Other files: clippy formatting

maqi reviewed (changes requested):
PutOnHold as `relay_client` is going to be removed after the reachability work got merged in.

vphongph committed:
chore(ant-node): add deprecation notices for relay_client feature

Relay nodes are being removed from the network. Add deprecation
comments on the relay_client struct field, builder method, examples,
and README documentation to inform developers this feature will no
longer be supported in a future release.
